dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/DatasetVersionDifferenceTest.java

```diff
@@ -0,0 +1,283 @@
+                dsDescriptionValueFieldType, dsDescriptionDateFieldType);
+        
+        DatasetFieldType subjectFieldType = MocksFactory.makeControlledVocabDatasetFieldType("subject", true, citationMetadataBlock,
+                "agricultural_sciences", "arts_and_humanities", "chemistry");
+        DatasetFieldType depositorFieldType = MocksFactory.makeDatasetFieldType("depositor", FieldType.TEXT, false, citationMetadataBlock);
+        DatasetFieldType dateOfDepositFieldType = MocksFactory.makeDatasetFieldType("dateOfDeposit", FieldType.TEXT, false, citationMetadataBlock);
+        
+        when(datasetFieldService.findByNameOpt(eq("title"))).thenReturn(titleFieldType);
+        when(datasetFieldService.findByNameOpt(eq("author"))).thenReturn(authorFieldType);
+        when(datasetFieldService.findByNameOpt(eq("authorName"))).thenReturn(authorNameFieldType);
+        when(datasetFieldService.findByNameOpt(eq("authorAffiliation"))).thenReturn(authorAffiliationFieldType);
+        when(datasetFieldService.findByNameOpt(eq("datasetContact"))).thenReturn(datasetContactFieldType);
+        when(datasetFieldService.findByNameOpt(eq("datasetContactName"))).thenReturn(datasetContactNameFieldType);
+        when(datasetFieldService.findByNameOpt(eq("datasetContactAffiliation"))).thenReturn(datasetContactAffiliationFieldType);
+        when(datasetFieldService.findByNameOpt(eq("datasetContactEmail"))).thenReturn(datasetContactEmailFieldType);
+        when(datasetFieldService.findByNameOpt(eq("dsDescription"))).thenReturn(dsDescriptionFieldType);
+        when(datasetFieldService.findByNameOpt(eq("dsDescriptionValue"))).thenReturn(dsDescriptionValueFieldType);
+        when(datasetFieldService.findByNameOpt(eq("dsDescriptionDate"))).thenReturn(dsDescriptionDateFieldType);
+        when(datasetFieldService.findByNameOpt(eq("subject"))).thenReturn(subjectFieldType);
+        when(datasetFieldService.findControlledVocabularyValueByDatasetFieldTypeAndStrValue(subjectFieldType, "Chemistry", false))
+            .thenReturn(subjectFieldType.getControlledVocabularyValue("chemistry"));
+        when(datasetFieldService.findControlledVocabularyValueByDatasetFieldTypeAndStrValue(subjectFieldType, "Agricultural Sciences", false))
+            .thenReturn(subjectFieldType.getControlledVocabularyValue("agricultural_sciences"));
+        when(datasetFieldService.findControlledVocabularyValueByDatasetFieldTypeAndStrValue(subjectFieldType, "Arts and Humanities", false))
+        .thenReturn(subjectFieldType.getControlledVocabularyValue("arts_and_humanities"));
+        
+        when(datasetFieldService.findByNameOpt(eq("depositor"))).thenReturn(depositorFieldType);
+        when(datasetFieldService.findByNameOpt(eq("dateOfDeposit"))).thenReturn(dateOfDepositFieldType);
+        
+        
+    }
+    
+    
+    // -------------------- TESTS --------------------
+    
+    @Test
```

Comment:
Can we split this into smaller bits? For primitive/compound fields, changed/unchanged versions ect.? 
There's too much going on for a unit test.

Reply:
For fields metadata it would be kind of complicated. We would have to create separate file with final metadata for each test. I can split cases with file changes.